matcher.go

```diff
@@ -63,7 +63,10 @@ func MatchesFilter(identifiable AttributeSpecifiable, filter *Filter, opts ...Ma
 				NotInComparator,
 				ContainComparator,
 				NotContainComparator:
-				return false, fmt.Errorf("elemental: unsuported comparator %q", translateComparator(filter.Comparators()[i]))
+				return false, &MatcherError{
+					kind:        KindUnsupportedComparator,
+					description: fmt.Sprintf("unsuported comparator %q", translateComparator(filter.Comparators()[i])),
+				}
```

Comment:
👀

Reply:
I would still prefer to not have a kind, but custom errors, wrapped in a `MatcherError`

Reply:
done, let me know what you think 🙏